src/coord/src/coord.rs

```diff
@@ -2431,6 +2430,18 @@ where
         Ok(())
     }
 
+    fn set_index_options(&mut self, id: GlobalId, options: Vec<IndexOption>) {
+        let index = self.indexes.get_mut(&id).expect("index known to exist");
```

Comment:
my reasoning earlier was that someone could interleave index creation / alteration and delete in such a way that we might get to this point and have the index be deleted. is that not actually possible? (don't totally remember my thought process offhand, not taking the time to read through the file rn)

Reply:
Indeed it is not! The correctness of the whole system hinges on the fact that the coordinator is single threaded and always applies the effects of a SQL query immediately after planning it. So the read-only copy of the catalog that the SQL planner uses to compute the plan is guaranteed to be the same catalog that the coordinator is applying updates to.

Reply:
ok i remember now i was going off of sinks -- but sinks have the asynchronous step where they create external state that breaks the order of updates guarantee